spec/validators/evidence_held_validator_spec.rb

```diff
@@ -0,0 +1,168 @@
+
+          it "has a meaningful error", :aggregate_failures do
+            expect(subject).to be_invalid
+            expect(subject.errors.messages_for(:evidence_held)).to include("Enter a '#/evidence_held' value for this participant.")
+          end
+        end
+
+        context "when `evidence_held` is present" do
+          let(:evidence_held) { "other" }
+
+          it "is valid" do
+            expect(subject).to be_valid
+          end
+
+          context "when `evidence_held` is invalid" do
+            let(:evidence_held) { "anything" }
+
+            it "has a meaningful error", :aggregate_failures do
+              expect(subject).to be_invalid
+              expect(subject.errors.messages_for(:evidence_held)).to include("Enter an available '#/evidence_held' type for this participant's event and course.")
+            end
+          end
+        end
+      end
+    end
+
+    context "Mentor participant" do
+      let(:declaration) { create(:mentor_participant_declaration) }
+      let(:participant_profile) { declaration.participant_profile }
+      let(:declaration_type) { declaration.declaration_type }
+      let(:evidence_held) { "other" }
+
+      subject { klass.new(participant_profile:, declaration_type:, evidence_held:) }
+
+      context "with valid params" do
+        it "is valid" do
+          expect(subject).to be_valid
+        end
+      end
+
+      context "when `declaration_type` is started" do
+        let(:declaration) { create(:ect_participant_declaration, declaration_type: "started") }
+
+        context "when `evidence_held` is not present" do
+          let(:evidence_held) { nil }
+
+          it "is valid" do
+            expect(subject).to be_valid
+          end
+        end
+
+        context "when `evidence_held` is present" do
+          let(:evidence_held) { "other" }
+
+          it "is valid" do
+            expect(subject).to be_valid
+          end
+
+          context "when `evidence_held` is invalid" do
+            let(:evidence_held) { "anything" }
+
+            it "has a meaningful error", :aggregate_failures do
+              expect(subject).to be_invalid
+              expect(subject.errors.messages_for(:evidence_held)).to include("Enter an available '#/evidence_held' type for this participant's event and course.")
+            end
+          end
+        end
+      end
+
+      context "when `declaration_type` is other than started or completed" do
+        let(:declaration) { create(:mentor_participant_declaration, declaration_type: "retained-1") }
+
+        context "when `evidence_held` is not present" do
+          let(:evidence_held) { nil }
+
+          it "has a meaningful error", :aggregate_failures do
+            expect(subject).to be_invalid
+            expect(subject.errors.messages_for(:evidence_held)).to include("Enter a '#/evidence_held' value for this participant.")
+          end
+        end
+
+        context "when `evidence_held` is present" do
```

Comment:
I wonder if we should fallback to the other validations in record declaration for declaration types for mentors being not supported, to avoid having an extra error over that one being returned 🤔